util_test.go

```diff
@@ -195,45 +195,45 @@ type InMemoryOutput struct {
 	deletesCounter int
 }
 
-func (out InMemoryOutput) Validate(secret *Secret, state secretState) bool {
+func (out *InMemoryOutput) Validate(secret *Secret, state secretState) bool {
 	_, present := out.Secrets[secret.Name]
 	// If it's in the map, it's valid - delete from the map to test on-disk invalidation behavior.
 	return present
 }
 
-func (out InMemoryOutput) Write(secret *Secret) (*secretState, error) {
+func (out *InMemoryOutput) Write(secret *Secret) (*secretState, error) {
 	out.Secrets[secret.Name] = *secret
 	out.writesCounter++
```

Comment:
The linter, when enabled, correctly pointed out that the updates to`writesCounter` and `deletesCounter` were no-ops since these methods were updating a copy of the `InMemoryOutput`, since we weren't taking `out` by pointer.